renn/data/datasets.py

```diff
@@ -201,13 +204,36 @@ def _preprocess(d):
     })
 
   # Load dataset.
-  dset = load_text_classification('snli', split, _preprocess, data_dir=data_dir)
-
-  # Apply custom filter.
-  if filter_fn is not None:
-    dset = dset.filter(filter_fn)
+  dset = load_text_classification('snli',
+                                  split,
+                                  _preprocess,
+                                  filter_fn,
+                                  data_dir=data_dir)
 
   # Pad remaining examples to the sequence length.
   dset = padded_batch(dset, batch_size, sequence_length)
 
   return dset
+
+
+def mnist(split,
```

Comment:
I assume this MNIST snippet is supposed to be part of a separate PR?

Reply:
Whoops, yeah.  That PR is open now